client/signup/steps/domains/index.jsx

```diff
@@ -753,7 +763,8 @@ export class RenderDomainsStep extends Component {
 			( suggestion && Boolean( suggestion.product_slug ) ) || domainCart?.length > 0;
 		const siteUrl =
 			suggestion &&
-			( isPurchasingItem
+			// If we have a free domain in the cart, we want to use it as the siteUrl
+			( isPurchasingItem && ! this.state.wpcomSubdomainSelected
```

Comment:
We must remove `.wordpress.com` on `siteUrl` to create a free site with a free suggestion domain_name.